### ISSUE TITLE
Support for non-primary key ids

### ISSUE THREAD

**Issue by devlato** [overlookmotel/sequelize-hierarchy#11]
Add support for customization of id field with additional option like `primaryKeyAttribute`, i.e.:
```javascript
module.exports = function(sequelize, DataTypes) {
  var fields, model, settings;
  fields = {
    id: {
      type: DataTypes.INTEGER.UNSIGNED,
      allowNull: false,
      primaryKey: true,
      autoIncrement: true,
      validate: {
        isInt: true
      }
    },
    createdAt: {
      type: 'TIMESTAMP'
    },
    updatedAt: {
      type: 'TIMESTAMP'
    },
    widgetType: {
      type: DataTypes.STRING,
      validate: {
        notEmpty: true
      }
    },
    parentId: {
      type: DataTypes.INTEGER.UNSIGNED,
      allowNull: true,
      references: 'widget_instance',
      referencesKey: 'id',
      hierarchy: {
        as: 'someId',
        childrenAs: 'content',
        primaryKeyAttribute: 'widgetInstancePublicId'
      },
      validate: {
        isInt: true
      }
    },
    hierarchyLevel: {
      type: DataTypes.INTEGER.UNSIGNED,
      allowNull: true
    },
    documentId: {
      type: DataTypes.INTEGER.UNSIGNED,
      references: 'document',
      referencesKey: 'id',
      validate: {
        isInt: true
      }
    },
    widgetInstancePublicId: {
      type: DataTypes.INTEGER.UNSIGNED,
      validate: {
        isInt: true
      }
    }
  };
  settings = {
    tableName: 'widget_instance'
  };
  model = sequelize.define('WidgetInstance', fields, settings);
  return model;
};
```

**Comment by overlookmotel** [comment]
Thanks for raising this issue and submitting a PR.

However, I'm having difficulty understanding why you'd want this. It only makes sense to me for `parentId` to reference the model's primary key. Can you explain?

Also, I'm a bit confused by the example you posted. If `parentId.hierarchy.primaryKeyAttribute` = `'widgetInstancePublicId'`, shoudn't `parentId.referencesKey` = `'widgetInstancePublicId'` too?

**Comment by devlato** [comment]
There can be some cases when parentId should be refering to some another field, i.e. to widgetInstancePublicId in the example above :-) 

> parentId.hierarchy.primaryKeyAttribute = 'widgetInstancePublicId', shoudn't parentId.referencesKey = 'widgetInstancePublicId'
Yeah, and there is a pull request covering that issue: 
https://github.com/overlookmotel/sequelize-hierarchy/pull/12

**Comment by overlookmotel** [comment]
OK, but can you give me an fuller explanation of why you'd want to do this? The idea of a hierarchy is that each "node" in the tree has a single parent, so why wouldn't the `parentId` refer to the primary key of that parent?

Concerning my question "if `parentId.hierarchy.primaryKeyAttribute` = `'widgetInstancePublicId'`, shoudn't `parentId.referencesKey` = `'widgetInstancePublicId'` too": I'm not clear from your answer whether there was a typo in the original code you posted (where `parentId.referencesKey` = `'id'`) or if I'm missing something.

Please help me to understand what you're trying to do...

**Comment by devlato** [comment]
Okay, i.e. there are three tables: `Rooms`, `Items` and `Properties`. 
`Items` table has an autoincrement primary key `id`, a composite key (`publicId`, `roomId`) where `publicId` is some value that is set from the outside (it's a real life case) and `roomId` is a foreign key to `Rooms`, and some value field (for example). 
`Properties` table has `id`, foreign key `itemId` to `Items`, and some value field too (for example).
So, let the `Items` table be a hierarchical structure, so there are additional fields — `parentId` and `hierarchyLevel`, but there is no information about item ids before they are saved (because `id` is an autoincrement field), and i.e. we want to bulk save all of them at once for speed optimization purposes, so `parentId` should be referenced to `publicId`. 
That's the real life case explained on synthetical example, so that's why this feature is required.

**Comment by devlato** [comment]
Hey, are you alive? :-)

**Comment by overlookmotel** [comment]
Fixed in v0.2.10

**Comment by overlookmotel** [comment]
NB `options.primaryKey` was implemented but it seems is not currently working. #67
